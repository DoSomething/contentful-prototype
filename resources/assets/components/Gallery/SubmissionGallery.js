import PropTypes from 'prop-types';
import React from 'react';
import Gallery from './Gallery';
import ReportbackItem from '../ReportbackItem';
import { makeHash } from '../../helpers';

class SubmissionGallery extends React.Component {
  static renderReportbackItem(submission) {
    // @TODO: Normalize data for uploaded RBs vs API retrieved RBs if possible...
    const key = makeHash(submission.media.uri || submission.media.filePreviewUrl);
    const url = submission.media.uri || submission.media.filePreviewUrl;

    return <ReportbackItem key={key} {...submission} url={url} reaction={null} basicDisplay />;
  }

  componentDidMount() {
    this.props.fetchUserReportbacks(this.props.userId, this.props.legacyCampaignId);
  }

  render() {
    const isFetching = this.props.submissions.isFetching;
    const items = this.props.submissions.items;

    return isFetching
      ? <div className="spinner -centered" />
      : <Gallery type="triad">
        {items.map(submission => SubmissionGallery.renderReportbackItem(submission))}
      </Gallery>;
  }
}

SubmissionGallery.propTypes = {
  fetchUserReportbacks: PropTypes.func.isRequired,
  legacyCampaignId: PropTypes.string.isRequired,
  submissions: PropTypes.shape({
    isFetching: PropTypes.bool,
    items: PropTypes.array,
  }).isRequired,
  userId: PropTypes.string.isRequired,
};

export default SubmissionGallery;
