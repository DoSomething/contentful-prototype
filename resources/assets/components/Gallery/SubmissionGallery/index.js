export SubmissionGallery from './SubmissionGallery';

export SubmissionGalleryContainer from './SubmissionGalleryContainer';

export default './SubmissionGallery';
