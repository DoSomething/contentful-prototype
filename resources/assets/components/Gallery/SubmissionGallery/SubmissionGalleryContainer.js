import { connect } from 'react-redux';

import SubmissionGallery from './SubmissionGallery';
import { fetchUserReportbacks } from '../../../actions';

/**
 * Provide state from the Redux store as props for this component.
 */
const mapStateToProps = state => ({
  legacyCampaignId: state.campaign.legacyCampaignId,
  submissions: state.submissions,
  userId: state.user.id,
});

/**
 * Provide pre-bound functions that allow the component to dispatch
 * actions to the Redux store as props for this component.
 */
const mapDispatchToProps = {
  fetchUserReportbacks,
};

/**
 * Export the container component.
 */
export default connect(mapStateToProps, mapDispatchToProps)(SubmissionGallery);
