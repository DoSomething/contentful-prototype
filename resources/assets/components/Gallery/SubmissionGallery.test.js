import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SubmissionGallery from './SubmissionGallery';

const mediaItem = {
  media: {
    uri: 'http://fakelink.com',
  },
};

test('it renders correctly with required props', () => {
  const component = shallow(
    <SubmissionGallery submissions={{
      isFetching: false,
      items: [mediaItem],
    }}
    />,
  );

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('it renders loader while fetching', () => {
  const component = shallow(
    <SubmissionGallery submissions={{
      isFetching: true,
      items: [mediaItem],
    }}
    />,
  );

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
