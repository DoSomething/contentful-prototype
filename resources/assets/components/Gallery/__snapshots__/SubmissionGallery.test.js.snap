// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly with required props 1`] = `
<Gallery
  type="triad"
>
  <ReportbackItem
    basicDisplay={true}
    firstName="A Doer"
    isFetching={false}
    media={
      Object {
        "uri": "http://fakelink.com",
      }
    }
    noun={
      Object {
        "plural": "items",
        "singular": "item",
      }
    }
    reaction={null}
    toggleReactionOff={[Function]}
    toggleReactionOn={[Function]}
    url="http://fakelink.com"
  />
</Gallery>
`;

exports[`it renders loader while fetching 1`] = `
<div
  className="spinner -centered"
/>
`;
